src/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,196 @@
+    checkedArray: Array<RadioState>;
+
+    static defaultProps = {
+        dataSource: [],
+        location: 'right',
+        disabled: false
+    };
+
+    constructor(props: RadioGroupProps) {
+        super(props);
+        this.checkedArray = [];
+        let flag = true;
+        if (this.props.children) {
+            for (let i = 0; i < this.props.children.length; i++) {
+                this.checkedArray.push(observable({checked: flag ? this.props.children[i].props.checked: false}));
+                if (this.props.children[i].props.checked) {
+                    flag = false;
+                }
+            }
+        } else if(this.props.dataSource) {
+            for (let i = 0; i < this.props.dataSource.length; i++) {
+                this.checkedArray.push(observable({checked: false}));
+            }
+        }
+        this.name = this.props.name ? this.props.name : 'name_' + counter++;
+    }
+
+    childrenOnClick(index: number) {
+        return (e: string) => {
+            this.checkedArray.forEach((value) => {
+                value.checked = false;
+            });
+            this.checkedArray[index].checked = true;
+            if (this.props.onChange) {
+                this.props.onChange(e);
+            }
+        };
+    }
+
+    createChildren(dataArray: any, isDataSource: boolean) {
+        const childArray: React.ReactNode[] = [];
+        for (let index = 0; index < dataArray.length; index++) {
+            const data = dataArray[index];
+            const props: RadioButtonProps = {
+                key: index,
+                value: '',
+                automationId: 'RADIO_BUTTON_' + index,
+                checked: this.checkedArray[index].checked,
+                onClick: this.childrenOnClick(index),
+                disabled: this.props.disabled,
+                location: this.props.location,
+                name: this.name};
+
+            if (isDataSource) {
+                props.value = data;
+                childArray.push(React.createElement(RadioButton, props));
+            } else {
+                if (data.type === RadioButton) {
+                    if ((data.props as RadioButtonProps).disabled) {
+                        props.disabled = true;
+                    }
+                    props.value = (data.props as RadioButtonProps).value;
+                    childArray.push(React.cloneElement(data as ReactElement<any>, props));
+                } else {
+                    childArray.push(React.cloneElement(data as ReactElement<any>, {key: index,
+                        checked: this.checkedArray[index].checked,
+                        onClick: action(this.childrenOnClick(index))}));
+                }
+            }
+        }
+        return childArray;
+    }
+
+    render() {
+        const dataArray = this.props.children ? this.props.children : this.props.dataSource;
+        let childArray: React.ReactNode[] = [];
+
+        if (this.props.dataSource!.length > 0) {
```

Comment:
I find it hard at this point to understand the rendering logic. It seems that dataArray prefers children over dataSource if they're provided, yet here it first of all checks for dataSource and populates childArray according to it...